File: translations/codenet/deepseek_commented/C2J/gpt-4o/Code/s858795471.java
import java.util.Scanner;

public class Main {
    public static int search(int[] A, int n, int key) {
        int i = 0;
        // Add the key to the end of the array to handle the case where the key is not found
        A[n] = key;
        // Loop until the key is found in the array
        while (A[i] != key) i++;
        // Return 1 if the key is found, otherwise return 0
        return i != n ? 1 : 0;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[] A = new int[10000 + 1];
        int q, key, sum = 0;

        // Read the array elements
        for (int i = 0; i < n; i++) {
            A[i] = scanner.nextInt();
        }
        // Read the number of queries
        q = scanner.nextInt();
        // Process each query
        for (int i = 0; i < q; i++) {
            key = scanner.nextInt();
            // Perform the search operation for the current query element
            if (search(A, n, key) == 1) sum++;
        }
        // Print the count of successful searches
        System.out.println(sum);
        
        scanner.close();
    }
}

// <END-OF-CODE>

File: translations/codenet/deepseek_commented/C2Go/gpt-4o/Code/s858795471.go
package main

import (
	"fmt"
)

func search(A []int, n, key int) bool {
	// Add the key to the end of the slice to handle the case where the key is not found
	A = append(A, key)
	i := 0
	// Loop until the key is found in the slice
	for A[i] != key {
		i++
	}
	// Return true if the key is found, otherwise return false
	return i != n
}

func main() {
	var n, q, key, sum int

	// Read the size of the array
	fmt.Scan(&n)
	A := make([]int, n)

	// Read the array elements
	for i := 0; i < n; i++ {
		fmt.Scan(&A[i])
	}

	// Read the number of queries
	fmt.Scan(&q)

	// Process each query
	for i := 0; i < q; i++ {
		fmt.Scan(&key)
		// Perform the search operation for the current query element
		if search(A, n, key) {
			sum++
		}
	}

	// Print the count of successful searches
	fmt.Println(sum)
}

// <END-OF-CODE>
